openid-4-verifiable-presentations-1_0.md

```diff
@@ -616,11 +624,324 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+Credential.
+
+Each entry in `credentials` MUST be an object with the following properties:
+
+`id`:
+: REQUIRED. A string identifying the Credential in the response and, if provided,
+the constraints in `credential_sets`. The value MUST be a non-empty string
+consisting of alphanumeric, underscore (`_`) or hyphen (`-`) characters.
+Within the Authorization Request, the same `id` MUST NOT
+be present more than once.
+
+`format`:
+: REQUIRED. A string that specifies the format of the requested
+Verifiable Credential. Valid Credential Format Identifier values are defined in
+Appendix A of [@!OpenID.VCI].
+
+`meta`: 
+: OPTIONAL. An object defining additional properties requested by the Verifier that
+apply to the metadata and validity data of the Credential. The properties of
+this object are defined per Credential Format in (#format_specific_properties). If omitted,
+no specific constraints are placed on the metadata or validity of the requested
+Credential.
+
+`claims`:
+: OPTIONAL. A non-empty array of objects as defined in (#claims_query) that specifies
+claims in the requested Credentials.
+
+`claim_sets`:
+: OPTIONAL. A non-empty array containing arrays of identifiers for
+elements in `claims`. The rules for selecting claims to send are defined in (#selecting_claims).
+
+Note that multiple Credential Queries in a request MAY request a presentation of the same Credential.
+
+## Credential Set Query {#credential_set_query}
+
+A Credential Set Query is an object representing a request for one or more credentials to satisfy
+a particular use case with the Verifier.
+
+Each entry in `credential_sets` MUST be an object with the following properties:
+
+`options`
+: REQUIRED: A non-empty array, where each value in the array is a list
+of Credential Query identifiers representing one set of Credentials that
+satisfies the use case. The value of each element in the `options` array is an
+array of identifiers which reference elements in `credentials`.
+
+`required`
+: OPTIONAL. A boolean which indicates whether this set of Credentials is required
+to satisfy the particular use case at the Verifier. If omitted, the default value is `true`.
+
+`purpose`
+: OPTIONAL. A string, number or object specifying the purpose of the
+query. This specification does not define a specific structure or specific values
+for this property. The purpose is intended to be used by the Verifier to
+communicate the reason for the query to the Wallet. The Wallet MAY use this
+information to show the user the reason for the request.
+
+## Claims Query {#claims_query}
+
+Each entry in `claims` MUST be an object with the following properties:
+
+`id`:
+: REQUIRED if `claim_sets` is present in the Credential Query; OPTIONAL otherwise. A string
+identifying the particular claim. The value MUST be a non-empty string
+consisting of alphanumeric, underscore (`_`) or hyphen (`-`) characters.
+Within the particular `claims` array, the same `id` MUST NOT
+be present more than once.
+
```

Comment:
As an alternative, couldn't we specify this object schema for each of the specified formats, similarly to what we do in VCI, instead of stating that it must be present on a format and absent every other formats.
I.e.
- `vc+sd-jwt`
    - `path` - REQUIRED (...)
- `mso_mdoc`
    -  `namespace`: REQUIRED (...)
    - `claim_name`: REQUIRED (...)    
     
E.g. this way we are forbidding `namespace` or `claim_name` to ever be used in a format which isn't based on mdoc.

Reply:
I agree, would seem cleaner to me to move these things to the Appendix B where each format is described an have a subsection there?

Reply:
+1 I opened #284 along these lines

Reply:
i thin that was where we started in the very first proposal but we ended up with this `meta` claim after discussions